cudamat/cudamat.py

```diff
@@ -1501,6 +1503,25 @@ def where(condition_mat, if_mat, else_mat, target=None):
     return target
 
 
+def filter(mat, filter, target=None):
```

Comment:
`filter` is a reserved keyword in Python. What about renaming the function to `correlate` ([as in scipy](http://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.correlate.html#scipy.ndimage.correlate)) and the argument to `kernel`?